src/commands/ext-uninstall.ts

```diff
@@ -29,72 +30,82 @@ export default new Command("ext:uninstall <extensionInstanceId>")
+
+      const extensionDeletionMessage =
+        `Here's what will happen when you uninstall ${clc.bold(instanceId)} from project ${clc.bold(
+          projectId
+        )}. Be aware that this cannot be undone.\n\n` +
+        `${serviceAccountMessage}` +
+        `${resourcesMessage}` +
+        `${artifactsMessage}`;
+
+      logger.info(extensionDeletionMessage);
       const confirmedExtensionDeletion = await promptOnce({
         type: "confirm",
         default: true,
-        message: extensionDeletionMessage,
+        message: "Are you sure?",
       });
       if (!confirmedExtensionDeletion) {
         return utils.reject("Command aborted.", { exit: 1 });
       }
-
-      const rolesMessage = _.get(instance, "config.source.spec.roles", []).length
-        ? " which had the following authorized roles in your project:\n" +
-          instance.config.source.spec.roles
-            .map((role: extensionsApi.Role) => `- ${role.role} \n`)
-            .join("")
-        : ". \n";
-      const serviceAccountDeletionMessage = `This extension used service account ${clc.bold(
-        instance.serviceAccountEmail
-      )} ${rolesMessage}Do you want to delete this service account?`;
-      confirmedServiceAccountDeletion = await promptOnce({
-        type: "confirm",
-        default: false,
-        message: serviceAccountDeletionMessage,
-      });
     }
 
     const spinner = ora.default(
-      `Uninstalling ${clc.bold(instanceId)}. This usually takes 1 to 2 minutes...`
+      ` ${clc.green.bold(logPrefix)}: uninstalling ${clc.bold(
+        instanceId
+      )}. This usually takes 1 to 2 minutes...`
     );
     spinner.start();
     try {
+      spinner.info();
+      spinner.text = ` ${clc.green.bold(logPrefix)}: deleting your extension instance's resources.`;
+      spinner.start();
       await extensionsApi.deleteInstance(projectId, instanceId);
-      if (confirmedServiceAccountDeletion || options.force) {
-        const saDeletionRes = await iam.deleteServiceAccount(
-          projectId,
-          instance.serviceAccountEmail
-        );
-        if (_.get(saDeletionRes, "body.error")) {
-          if (_.get(saDeletionRes, "body.error.code") === 404) {
-            spinner.succeed(
-              ` ${clc.green.bold(logPrefix)}: service account ${clc.bold(
-                instance.serviceAccountEmail
-              )} was previously deleted.`
-            );
-          } else {
-            throw new FirebaseError("Unable to delete service account", {
-              original: saDeletionRes.body.error,
-            });
-          }
-        } else {
+      spinner.succeed(
+        ` ${clc.green.bold(logPrefix)}: deleted your extension instance's resources.`
+      );
+      spinner.text = ` ${clc.green.bold(
+        logPrefix
+      )}: deleting your extension instance's service account.`;
+      spinner.start();
+      const saDeletionRes = await iam.deleteServiceAccount(projectId, instance.serviceAccountEmail);
```

Comment:
It would be faster to kick off service account deletion at the same time as deleting instance with `await Promise.all`

Reply:
I think the original intent of doing these synchronously is to ensure that we don't end up in a state where the SA delete worked but the instance deletion didn't, since that would leave the extension resources running in a broken state.

Reply:
Ok good point, let's leave as-is.